src/shipit_uplift/shipit_uplift/coverage_by_changeset_impl.py

```diff
@@ -0,0 +1,196 @@
+import json
+import requests
+import whatthepatch
+from config import CoverageService, COVERAGE_SERVICE
+
+
+github_commits = {}
+def get_github_commit(mercurial_commit):
+    if mercurial_commit not in github_commits:
+        url = 'https://api.pub.build.mozilla.org/mapper/gecko-dev/rev/hg/%s'
+        r = requests.get(url % mercurial_commit)
+
+        github_commits[mercurial_commit] = r.text.split(' ')[0]
+
+    return github_commits[mercurial_commit]
+
+
+coverage_data = {}
+def get_coverage(changeset, filename):
+    if COVERAGE_SERVICE == CoverageService.COVERALLS:
```

Comment:
I would create a base class and extends into 3 different children classes to avoid having various if blocks within each function to handle each coverage service.

Reply:
I will do a refactoring in a follow-up.

Reply:
Well, since I was flying and couldn't sleep, I've done it now.